woodwork/logical_types.py

```diff
@@ -76,7 +76,7 @@ def validate(self, series, return_invalid_values=True):
         specific validation, as required."""
         valid_dtype = self._get_valid_dtype(type(series))
         if valid_dtype != str(series.dtype):
-            raise ValueError(
+            raise TypeValidationError(
                 f"Cannot initialize Woodwork. Series dtype '{series.dtype}' is "
                 f"incompatible with {self.type_string} dtype. Try converting series "
                 f"dtype to '{valid_dtype}' before initializing or use the "
```

Comment:
Let's change the message here at the LogicalType.validate method to say

f"Series dtype '{series.dtype}' is incompatible with {self.type_string} dtype."
So that it can be used outside of Woodwork initialization